core/src/main/scala/cats/collections/Diet.scala

```diff
@@ -233,10 +234,18 @@ sealed abstract class Diet[A] {
     case DietNode(_, _, r) => r.max
   }
 
-  @deprecated("the semantics of this method are unclear and unspecified, avoid!", "0.8.0")
-  def map[B: Discrete: Order](f: A => B): Diet[B] = this match {
+  //  @deprecated("the semantics of this method are unclear and unspecified, avoid!", "0.8.0")
```

Comment:
I strongly disagree with un-deprecating this. Mapping may destroy the invariants.